crates/chia-bls/src/bls_cache.rs

```diff
@@ -88,6 +113,25 @@ impl BlsCache {
         let hash: [u8; 32] = hasher.finalize();
         self.cache.lock().expect("cache").put(hash, gt);
     }
+
+    pub fn evict<Pk, Msg>(&mut self, pks_msgs: impl IntoIterator<Item = (Pk, Msg)>)
+    where
+        Pk: Borrow<PublicKey>,
+        Msg: AsRef<[u8]>,
+    {
+        let mut hashes_to_remove = HashSet::new();
+        for (pk, msg) in pks_msgs {
+            let mut hasher = Sha256::new();
+            let mut aug_msg = pk.borrow().to_bytes().to_vec();
+            aug_msg.extend_from_slice(msg.as_ref());
+            hasher.update(&aug_msg);
+            let hash: [u8; 32] = hasher.finalize();
+            hashes_to_remove.insert(hash);
+        }
+        let mut c = self.cache.lock().expect("cache");
+        c.items.retain(|h, _| !hashes_to_remove.contains(h));
```

Comment:
this is a hash map. surely it's faster to remove items individually, using its own index, rather than iterating over all items. Did you benchmark that?
This looks a bit unintuitive I think. If you benchmarked and this is faster, please leave a comment saying so